src/implementation/Client/HTTPClient/HTTPClient.ts

```diff
@@ -34,8 +34,9 @@ export default class HTTPClient implements IClient {
 
   private readonly httpAgent;
   private readonly httpsAgent;
+  private static _instance: HTTPClient;
 
-  constructor(
+  private constructor(
```

Comment:
As discussed offline, the `HttpClient` object should not be cached. Rather cache the actual HTTP client, `fetch` in this case. Caching the `HttpClient` will return the same instance for different host/port/options combination.